src/js/CoachmarkListener.js

```
@@ -51,143 +37,175 @@ export default class CoachmarkListener {
-			this.getDisplayCoachmark(masterpieceId);
 		});
 	}
 
 	/**
 	 * Sets up the next listener
 	 **/
-	setupNextListener(masterpieceId) {
-		const cmIds = this.cmState[masterpieceId].cmIds;
-
+	_setupNextListener() {
 		document.addEventListener('o-cm-next-clicked', (event) => {
-			const index = this.cmState[masterpieceId].index;
-			if (masterpieceId !== event.data.id || index + 1 === cmIds.length) {
-				return;
+			try {
+				const state = JSON.parse(event.data.id);
+				if (state.index + 1 === state.cmIds.length) {
+					return;
+				}
+				state.index++;
+				this._getDisplayCoachmark(state);
+			} catch (e) {
+				this._handleError(e);
 			}
-			this.cmState[masterpieceId].index++;
-			this.getDisplayCoachmark(masterpieceId);
 		});
 	}
 
 
+
 	/**
 	 * Gets data from the API and displays a coachmark on the correct page
 	 **/
-	getDisplayCoachmark(masterpieceId) {
-		const cmIds = this.cmState[masterpieceId].cmIds;
-		const index = this.cmState[masterpieceId].index;
-		const cmId = cmIds[index];
-
-		this.coachmarkApi.getCoachmark(cmId).then((result) => {
-			// Redirect if this coachmark ID is meant to display on a different page
-			if (this.redirectIfNewUri(result.uri, masterpieceId)) {
-				return;
-			}
-
-			// Auto-populating options to simplify the coachmark payload
-			const options = result.options;
-			options.id = masterpieceId;
-			options.currentCM = parseInt(index) + 1;
-			options.totalCM = cmIds.length;
-
-			// Display the coach mark
-			new Coachmark(document.getElementById(result.element), options, () => {
-				this.notificationApi.markAsRead(this.cmState[masterpieceId].userNotificationId);
+	_getDisplayCoachmark(state) {
+		const cmId = state.cmIds[state.index];
+
+		this.coachmarkApi.getCoachmark(cmId)
+			.then((coachmark) => {
+				// Redirect if this coachmark ID is meant to display on a different page
+				if (this._redirectIfNewUri(coachmark.uri, state)) {
+					return;
+				}
+
+				// Tick hit counter if first visit
+				try {
```

At line 146 the 'true' value being sent to _handleError tells it to silently swallow the error, no coachmark will be displayed